pyext/src/analysis_trajectories.py

```diff
@@ -42,25 +43,74 @@ class AnalysisTrajectories(object):
     def __init__(self,
                  out_dirs,
                  dir_name = 'run_',
-                 out_name = 'output',
-                 analysis_dir = 'analysis/',
-                 nproc=6,
-                 nskip=200):
+                 analysis_dir = 'analysis',
+                 detect_equilibration=True,
+                 burn_in_fraction=0.02,
+                 nskip=20,
+                 nproc=1,
+                 plot_fmt='pdf'):
+        
+        '''
+        Analyze the ensemble of models obtained after structural sampling by
+        filtering out the bad scoring models, and clustering the good scoring
+        models according to one or multiple score terms, thereby obtaining
+        a set of representative good scoring models that can be used to
+        subsequently compute sampling precision and model precision [using the
+        sampcon module].
+
+        @param out_dirs: list of directories containing the trajectory (rmf
+        files and stat files) for each independent run.
+        
+        @param dir_name: prefix for all directories containing structural 
+        sampling runs. Usually these are named as <prefix>1, <prefix>2, etc.
+        Default is "run_"
+        
+        @analysis_dir: name of output directory where all analysis results will
+        be stored. Default is "analysis".
+        
+        @detect_equilibration: if True, the equilibration period is rigorously
+        determined from the trajectory of all scores using statistical inefficiency measures. Default is True.
+        
+        @burn_in_fraction: a <burn_in_fraction> fraction of frames are always discarded from the beginning of the trajectory, when calculating statistics. Default is 0.02 (i.e. 2 % of the total trajectory size).
+        
+        @nskip: skips every <nskip> frames when calculating statistics.
+        Default is 200.
+        
+        @nproc: number of processors for parallel execution (uses the python multiprocess module). Default is 1.
+        
+        @plot_fmt: file extensions for all plots created. Default is pdf.
+        '''
 
-        self.out_dirs = out_dirs
+        self.out_dirs = [os.path.abspath(d) for d in out_dirs]
         self.dir_name = dir_name
-        self.analysis_dir = analysis_dir
+        self.analysis_dir = os.path.abspath(analysis_dir)
+        self.detect_equilibration = detect_equilibration
         self.nproc = nproc
         self.nskip = nskip
+        self.burn_in_frac = burn_in_fraction
+
+        # check if plot_fmt is supported
+        fig = pl.figure()
+        supported_fmts = list(plt.gcf().canvas.get_supported_filetypes().keys())
+        if plot_fmt not in supported_fmts:
+            raise IOError("plot_fmt not found in supported matplotlib file types:", supported_fmts)
```

Comment:
IOError is an odd error to raise here. I'd expect something like KeyError normally.

Reply:
fixed.